### PR TITLE
Allow custom verifier in Credentials.Provided

### PR DESCRIPTION
**Pull request by andrewrigas**
References #1796

* Create provideVerify method
* Update tests in SecurityDirectivesExamplesSpec.scala

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @andrewrigas,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala [hunk 1]
```diff
@@ -302,17 +302,31 @@ object Credentials {
      * See also [[EnhancedString#secure_==]], for more information.
      */
     def verify(secret: String): Boolean = verify(secret, x => x)
+
+    /**
+     * Compares with custom 'verifier' the received secret part of the Credentials.
+     * Use of this method only if custom String equality testing is required not recommended.
```
Comment (johanandren) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala:308]: Not sure if I parse this sentence correctly, is it:
```suggestion
     * Use of this method only if custom String equality testing is required, not recommended.
```
?

File: docs/src/test/scala/docs/http/scaladsl/server/directives/SecurityDirectivesExamplesSpec.scala [hunk 2]
```diff
@@ -10,144 +10,176 @@ import akka.http.scaladsl.server.Route
...
 import scala.concurrent.Future
 
 class SecurityDirectivesExamplesSpec extends RoutingSpec with CompileOnlySpec {
 
+  def verifier(secret: String)(receivedSecret: String): Boolean = secret == receivedSecret
+
   "authenticateBasic-0" in {
     //#authenticateBasic-0
-    def myUserPassAuthenticator(credentials: Credentials): Option[String] =
+    def myUserPassAuthenticatorVerify(credentials: Credentials): Option[String] =
```
Comment (johanandren) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/SecurityDirectivesExamplesSpec.scala:23]: This is pulled into documentation, I'd expect we want to keep it unchanged and only add new test cases/examples for the new `provideVerify`?

Comment (andrewrigas) [reply docs/src/test/scala/docs/http/scaladsl/server/directives/SecurityDirectivesExamplesSpec.scala:23]: Yes, I am not sure where should I add new test for this, do we need to update the documentation test or is there any other file that test those SecurityDirectives?

Comment (andrewrigas) [reply docs/src/test/scala/docs/http/scaladsl/server/directives/SecurityDirectivesExamplesSpec.scala]: I think I found it, there wasn't any test for verify function I added one. Let me know if its all good.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**